### PR TITLE
Mocha unit tests - utilities and default options schema check

### PR DESCRIPTION
**Pull request by chefArchitect**
I have created an initial set of Mocha tests covering two sections:

- default options validity based on a schema
- unit test for **amqp-util** module

More unit tests will follow for the rest of the modules.

### CODE REVIEW

**Comment by nfantone** [comment]
@chefArchitect Want me to review and merge? Or shall I wait until you finish pushing changes?

**Comment by chefArchitect** [comment]
I suggest we do this in small iterations.  Check this one first and in the following days I'll be pushing more updates.

**Comment by nfantone** [comment]
@chefArchitect Reviewing this now.

File: examples/client.js [hunk 1]
```diff
@@ -14,5 +14,7 @@ setInterval(function() {
   client.act('role:create', {
     max: 100,
     min: 25
-  }, console.log);
+  }, function(err, msg) {
```
Comment (nfantone) [inline examples/client.js]: Would you mind not mixing stylistic changes with functional changes on the PR? :rabbit:

Comment (chefArchitect) [inline examples/client.js]: Sorry @nfantone - my bad.  Forgot to rollback this one.

File: test/lib/amqp-util.test.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const chai = require('chai')
```
Comment (nfantone) [inline test/lib/amqp-util.test.js]: Please, follow the Eslint rules enforced for the project. As we didn't have tests before, currently the `gulp eslint` task is't linting the new `test` directory.

But if you update `gulpfile.js:L17` with: 

```js
gulp.src([].concat(config.paths.src, config.paths.test))
```

and run `gulp eslint`, you'll see a total of ✖ 79 problems (75 errors, 4 warnings).

Could you fix those for me, please?

Comment (chefArchitect) [inline test/lib/amqp-util.test.js]: Sure, np.

**Comment by chefArchitect** [comment]
@nfantone Just pushed all latest unit tests + linting issue fixes.  If you want me to write tests that cover Hooks related functionality, let me know.

**Comment by nfantone** [comment]
Good stuff.

I'll take a look, ASAIC.

**Comment by chefArchitect** [comment]
Hi @nfantone.  Are you planning to merge this PR anytime soon or you want me to close it?  Didn't have any feedback from you for a while now.

**Comment by nfantone** [comment]
I'm sorry for the long silence. I've been unusually busy lately.

I'm all up for this. I'll try to review it before tomorrow.

**Comment by chefArchitect** [comment]
Cool! Thanks.

**Comment by nfantone** [comment]
@chefArchitect Ok, so first of all: _amazing_ work. Thank you very much for your efforts and contribution.

Everything LGTM. I can't ask you to do anything more than what you already did.

Just two lil' comments before merging this.

1. I know the IDE should probably bug you about it (I'm looking at you, Atom) - but, following the general `eslint-config-seneca` rules on `gulpfile.js` is not very _graceful_ to the eye. Using `Gulp` instead of `gulp` looks just plain silly. As long as `gulp validate` passes, everything else is fine to me. So, could you revert that, please?
2. Would you please rebase your PR branch? There have been two new commits to `develop` since.

**Comment by chefArchitect** [comment]
@nfantone Ok, we're cool now.  Synced the 'develop' branch and reverted the gulp config, excluding it also from linting.  Sorry for the inconvenience.

**Comment by nfantone** [comment]
There, done.

@chefArchitect Thanks again. I'll update the docs to add a contributors list with your name on it.

**Comment by chefArchitect** [comment]
@nfantone If you come by Athens anytime soon, let me know :)

**Pull request merged**